src/ui/components/search/autocompletecomponent.js

```diff
@@ -179,6 +187,19 @@ export default class AutoCompleteComponent extends Component {
       this._isOpen = true;
     }
 
+    // show custom prompts if it's an empty search or levenshtein match on a non empty search
+    if (this.customPrompts && data.sections) {
+      const hasCustomPrompts = data.sections.find(result => result.isCustom);
+      const matchedCustomPrompts = this.customPrompts.filter(prompt =>
+        isCloseMatch(prompt, queryInputEl.value)
+      );
+      if (!hasCustomPrompts && matchedCustomPrompts.length > 0) {
+        const dataWithCustomPrompts = this.combineResultsWithCustomPrompts(data, matchedCustomPrompts);
+        this.core.storage.set(`${StorageKeys.AUTOCOMPLETE}.${this.name}`, dataWithCustomPrompts);
```

Comment:
Is this kind of like doing a useLayoutEffect in react, where we update the state and trigger a re-render?

Reply:
yea, we update the state to include the custom prompts section (this is needed so keyboard navigation and query submission registers the right sections and indexes), and the return statement following this line ensures the old data doesn't get render.